docs/standard/serialization/system-text-json/snippets/system-text-json-how-to/csharp/Utf8ReaderCopyString.cs

```diff
@@ -0,0 +1,32 @@
+using System.Buffers;
+using System.Text.Json;
+
+namespace SystemTextJsonSamples
+{
+    public class Utf8ReaderCopyString
+    {
+        public static void Run()
+        {
+            // <Snippet1>
+            var reader = new Utf8JsonReader( /* jsonReadOnlySpan */ );
+
+            int valueLength = reader.HasValueSequence
+                ? checked((int)reader.ValueSequence.Length)
+                : reader.ValueSpan.Length;
+
+            char[] buffer = ArrayPool<char>.Shared.Rent(valueLength);
+            int charsRead = reader.CopyString(buffer);
+            ReadOnlySpan<char> source = buffer.AsSpan(0, charsRead);
```

Comment:
@eiriktsarpalis I changed this from buffer.Slice to buffer.AsSpan. Is that correct?

Reply:
Yes, I think the previous snippet wouldn't compile actually.